libs/components/text-editor/src/lib/modules/text-editor/url-modal/text-editor-url-modal.component.ts

```diff
@@ -73,6 +76,13 @@ export class SkyTextEditorUrlModalComponent {
   readonly #modalInstance = inject(SkyModalInstance);
 
   constructor() {
+    if (this.#modalContext.linkWindowOptions?.length === 1) {
```

Comment:
When referencing a property that's multiple levels deep, it's good to assign it to a variable first. This is slightly more efficient and results in less code once it's minified for production. You can also combine the two `if` statements into one and check the first item instead of using `includes()`.

```
const linkWindowOptions = this.#modalContext.linkWindowOptions;

if (linkWindowOptions?.length === 1 && linkWindowOptions[0] === 'new') {
}
```

Reply:
Assigned the variable first and then added an `if-else` to set the correct target depending whether the option was 'new' or 'existing'.